src/univariate/continuous/lindley.jl

```diff
@@ -0,0 +1,180 @@
+"""
+    Lindley(θ)
+
+The one-parameter *Lindley distribution* with shape `θ > 0` has probability density
+function
+
+```math
+f(x; \\theta) = \\frac{\\theta^2}{1 + \\theta} (1 + x) e^{-\\theta x}, \\quad x > 0
+```
+
+It was first described by Lindley[^1] and was studied in greater detail by Ghitany
+et al.[^2]
+Note that `Lindley(θ)` is a mixture of an `Exponential(θ)` and a `Gamma(2, θ)` with
+respective mixing weights `p = θ/(1 + θ)` and `1 - p`.
+
+[^1]: Lindley, D. V. (1958). Fiducial Distributions and Bayes' Theorem. Journal of the
+      Royal Statistical Society: Series B (Methodological), 20(1), 102–107.
+[^2]: Ghitany, M. E., Atieh, B., & Nadarajah, S. (2008). Lindley distribution and its
+      application. Mathematics and Computers in Simulation, 78(4), 493–506.
+"""
+struct Lindley{T<:Real} <: ContinuousUnivariateDistribution
+    θ::T
+
+    Lindley{T}(θ::T) where {T} = new{T}(θ)
+end
+
+function Lindley(θ::Real; check_args::Bool=true)
+    @check_args Lindley (θ, θ > zero(θ))
+    return Lindley{typeof(θ)}(θ)
+end
+
+Lindley(θ::Integer; check_args::Bool=true) = Lindley(float(θ); check_args=check_args)
+
+Lindley() = Lindley{Float64}(1.0)
+
+Base.convert(::Type{Lindley{T}}, d::Lindley) where {T} = Lindley{T}(T(shape(d)))
+Base.convert(::Type{Lindley{T}}, d::Lindley{T}) where {T} = d
+
+@distr_support Lindley 0.0 Inf
+
+### Parameters
+
+shape(d::Lindley) = d.θ
+params(d::Lindley) = (shape(d),)
+partype(::Lindley{T}) where {T} = T
+
+### Statistics
+
+mean(d::Lindley) = (2 + d.θ) / d.θ / (1 + d.θ)
+
+var(d::Lindley) = 2 / d.θ^2 - 1 / (1 + d.θ)^2
+
+skewness(d::Lindley) = 2 * @evalpoly(d.θ, 2, 6, 6, 1) / @evalpoly(d.θ, 2, 4, 1)^(3//2)
+
+kurtosis(d::Lindley) = 3 * @evalpoly(d.θ, 8, 32, 44, 24, 3) / @evalpoly(d.θ, 2, 4, 1)^2 - 3
+
+mode(d::Lindley) = d.θ < 1 ? (1 - d.θ) / d.θ : zero(d.θ)
```

Comment:
If theta is e.g. an integer, this would not be type stable. Maybe safer:
```suggestion
mode(d::Lindley) = max(0, (1 - d.θ) / d.θ)
```